packages/colorpickers/src/elements/ColorPicker/reducer.ts

```diff
@@ -0,0 +1,298 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { parseToHsl, parseToRgb, rgb as rgbToString } from 'polished';
+import { hsv2hsl, rgbToHsl, hsl2rgb } from '../../utils/conversion';
+
+export interface IColorPickerState {
+  hex: string;
+  hue: number;
+  saturation: number;
+  lightness: number;
+  red: number;
+  green: number;
+  blue: number;
+  alpha: number;
+  redInput: number | string;
+  greenInput: number | string;
+  blueInput: number | string;
+  alphaInput: number | string;
+}
+
+export const SATURATION_CHANGE = 'saturation change';
+export const HUE_CHANGE = 'hue slider change';
+export const ALPHA_SLIDER_CHANGE = 'alpha slider change';
+export const HEX_CHANGE = 'hex change';
+export const RED_CHANGE = 'red change';
+export const GREEN_CHANGE = 'green change';
+export const BLUE_CHANGE = 'blue change';
+export const ALPHA_CHANGE = 'alpha change';
+
+export interface IHSVColor {
+  h: number;
+  s: number;
+  v: number;
+}
+
+export interface IHSLColor {
+  h: number;
+  s: number;
+  l: number;
+  a?: number;
+}
+
+export interface IRGBColor {
+  red: number;
+  green: number;
+  blue: number;
+  alpha?: number;
+}
+
+export interface ISaturationChange {
+  type: typeof SATURATION_CHANGE;
+  payload: IHSVColor;
+}
+
+export interface IHueSliderChange {
+  type: typeof HUE_CHANGE;
+  payload: string;
+}
+
+export interface IAlphaSliderChange {
+  type: typeof ALPHA_SLIDER_CHANGE;
+  payload: string;
+}
+
+export interface IHexChange {
+  type: typeof HEX_CHANGE;
+  payload: string;
+}
+
+export interface IRedChange {
+  type: typeof RED_CHANGE;
+  payload: string;
+}
+
+export interface IGreenChange {
+  type: typeof GREEN_CHANGE;
+  payload: string;
+}
+
+export interface IBlueChange {
+  type: typeof BLUE_CHANGE;
+  payload: string;
+}
+
+export interface IAlphaChange {
+  type: typeof ALPHA_CHANGE;
+  payload: string;
+}
+
+export type ColorPickerActionTypes =
+  | ISaturationChange
+  | IHueSliderChange
+  | IAlphaSliderChange
+  | IHexChange
+  | IRedChange
+  | IGreenChange
+  | IBlueChange
+  | IAlphaChange;
```

Comment:
There are many ways to type these reducer actions, but let's try and stick with the format provided in [react-datepickers](https://github.com/zendeskgarden/react-components/blob/main/packages/datepickers/src/elements/Datepicker/utils/datepicker-reducer.ts#L85-L93)

This will cut down on a lot of the TS boilerplate.